### PR TITLE
test: private only schematics test

### PR DESCRIPTION
**Pull request by kierramarie**
### Description

A private only secrets manager test has been added to the pr_test.go.

Issue: https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager-private-cert-engine/issues/192

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ These components make up the `private_cert` secrets type. The module also signs
 * [terraform-ibm-secrets-manager-private-cert-engine](#terraform-ibm-secrets-manager-private-cert-engine)
 * [Examples](./examples)
     * [Example with Secrets Manager and the private certificates engine](./examples/default)
+    * [Example with Secrets Manager and the private certificates engine](./examples/private)
```
Comment (alex-reiff) [inline README.md:25]: more descriptive description

File: examples/private/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Example with Secrets Manager and the private certificates engine
```
Comment (alex-reiff) [inline examples/private/README.md:1]: more descriptive description

**Review by alex-reiff — CHANGES_REQUESTED** [review]

**Review by alex-reiff — DISMISSED** [review]

File: examples/private/version.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= v1.3.0, <1.7.0"
```
Comment (ocofaigh) [inline examples/private/version.tf:2]: We are no longer restricting upper limit for terraform version, so just change this to `">= 1.3.0"`

File: examples/private/version.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  required_providers {
+    ibm = {
+      source  = "IBM-Cloud/ibm"
+      version = "= 1.62.0"
```
Comment (ocofaigh) [inline examples/private/version.tf:6]: The default example is already running the lowest supported IBM provider version, so this example should run on latest. So change this to `>= 1.62.0`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: tests/pr_test.go [hunk 5]
```diff
@@ -44,3 +45,33 @@ func TestRunUpgradeExample(t *testing.T) {
 		assert.NotNil(t, output, "Expected some output")
 	}
 }
+
+func TestPrivateInSchematics(t *testing.T) {
+	t.Parallel()
+
+	const region = "us-south"
```
Comment (ocofaigh) [inline tests/pr_test.go:52]: Why hard code to us-south ? The extended quotas we have are per region, so we need to spread our tests around to all the regions in https://github.com/terraform-ibm-modules/common-dev-assets/blob/main/common-go-assets/cloudinfo-region-secmgr-prefs.yaml

File: tests/pr_test.go [hunk 6]
```diff
@@ -44,3 +45,33 @@ func TestRunUpgradeExample(t *testing.T) {
...
+		Prefix:  "sm-prv-crt-eng-private",
+		TarIncludePatterns: []string{
+			"*.tf",
+			testLocation + "/*.tf",
+		},
+		ResourceGroup:          resourceGroup,
+		TemplateFolder:         testLocation,
+		Tags:                   []string{"test-schematic"},
+		DeleteWorkspaceOnFail:  false,
+		WaitJobCompleteMinutes: 80,
```
Comment (ocofaigh) [inline tests/pr_test.go:65]: You should use `BestRegionYAMLPath: "../common-dev-assets/common-go-assets/cloudinfo-region-secmgr-prefs.yaml"` to ensure the test uses a region from this file - @toddgiguere this works for schematics test right?

Comment (toddgiguere) [reply tests/pr_test.go:65]: Yes, that will still work for the schematics selector.

**Comment by ocofaigh** [comment]
@kierramarie Can you resolve conflicts please?

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**